### PR TITLE
Fixed #28290 -- Enabled Sphinx auto-label generation for title and section headers.

### PR DESCRIPTION
**Pull request by abhijeetviswa**
Certain sections don't have labels which makes it difficult to refer them
from other Sphinx documents using InterSphinx. The AutoSectionLabel does
what it says.

Asked for opinion regarding a particular setting of the extension here: https://groups.google.com/forum/#!topic/django-developers/sRfq48rD-Ao

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -38,11 +38,18 @@
 # coming with Sphinx (named 'sphinx.ext.*') or your custom ones.
 extensions = [
     "djangodocs",
-    'sphinx.ext.extlinks',
+    "sphinx.ext.extlinks",
     "sphinx.ext.intersphinx",
     "sphinx.ext.viewcode",
+    "sphinx.ext.autosectionlabel"
```
Comment (felixxm) [inline docs/conf.py]: trailing comma

**Comment by abhijeetviswa** [comment]
Thank you. I'll create a discussion of the same on Sphinx's mailing list. I'm assuming such a change would break people who already use the extension.

File: docs/conf.py [hunk 2]
```diff
@@ -38,11 +38,18 @@
 # coming with Sphinx (named 'sphinx.ext.*') or your custom ones.
 extensions = [
     "djangodocs",
-    'sphinx.ext.extlinks',
+    "sphinx.ext.extlinks",
```
Comment (felixxm) [inline docs/conf.py]: This change is unrelated so we can revert it.

Comment (abhijeetviswa) [reply docs/conf.py]: The quotes were inconsistent in that list. Thought it would be better to make them all double quotes.

Comment (felixxm) [reply docs/conf.py]: I know but we shouldn't mix unrelated cleanups with bugfixes.
